tests/system_tests_qdstat.py

```diff
@@ -66,24 +66,33 @@ def test_help(self):
 
     def test_general(self):
         out = self.run_qdstat(['--general'], r'(?s)Router Statistics.*Mode\s*Standalone')
-        self.assertTrue("Connections                      1" in out)
-        self.assertTrue("Nodes                            0" in out)
-        self.assertTrue("Auto Links                       0" in out)
-        self.assertTrue("Link Routes                      0" in out)
-        self.assertTrue("Worker Threads                   1" in out)
-        self.assertTrue("Router Id                        QDR.A" in out)
-        self.assertTrue("Mode                             standalone" in out)
+
+        self.assertTrue(re.match(r"(.*)\bConnections\b[ \t]+\b1\b(.*)",
+                                 out, flags=re.DOTALL) is not None, out)
```

Comment:
There's `assertRegexpMatches` in Python 2.7 and unittest2. It does `re.search()` which is a better fit here (avoids having to put `.*` at the beginning and the end) and should give better assertion failure messages. https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRegex

Reply:
assertRegexpMatches gives a deprecation warning on Python 3.8 since it is replaced by assertRegex. I cannot use assertRegex because it fails on RHEL 6.

Reply:
That deprecation warning IMO deserves a targetted suppression there, but that's just my view.

I never really used the 'warnings.filterwarnings(action, ...', but it looks viable.